### PR TITLE
Add details about AddRequestedClaims

### PR DESCRIPTION
**Pull request by josephdecock**
Add more detail about when to use AddRequestedClaims, how it filters, and why.

Also, minor copy-editing of the claims page.

### CODE REVIEW

**Comment by josephdecock** [comment]
Resolves #260

File: IdentityServer/v6/docs/content/fundamentals/claims.md [hunk 1]
```diff
@@ -7,63 +7,77 @@ weight: 45
...
-Whenever IdentityServer creates tokens for a user, it invokes the registered profile service with a [context]({{< ref "/reference/services/profile_service#duendeidentityservermodelsprofiledatarequestcontext" >}}) that presents detailed information about the current token request, including
+Whenever IdentityServer needs the claims for a user, it invokes the registered profile service with a [context]({{< ref "/reference/services/profile_service#duendeidentityservermodelsprofiledatarequestcontext" >}}) that presents detailed information about the current request, including
 
-* the identity of the client who is requesting the token
+* the client that is making the request
 * the identity of the user
-* what type of token is requested
-* the requested claim types according to the definition of the requested resources
+* the type of the request (access token, id token, or userinfo)
+* the requested claim types, which are the claims types associated with requested identity resource scopes
```
Comment (brockallen) [inline IdentityServer/v6/docs/content/fundamentals/claims.md:17]: not all requests to the profile service are for claim types defined by "IdentityResources", right? they could be claim types defined by ApiScopes or ApiResources too.

File: IdentityServer/v6/docs/content/fundamentals/claims.md [hunk 2]
```diff
@@ -7,63 +7,77 @@ weight: 45
...
 
-You can use different strategies to determine which claims you want to emit based on that information
+### Strategies for Emitting Claims
+You can use different strategies to determine which claims to emit based on the information in the profile context.
 
-* always emit certain claims (because they are an integral part of the user identity and needed in scenarios)
+* emit claims based on the requested identity resources
```
Comment (brockallen) [inline IdentityServer/v6/docs/content/fundamentals/claims.md:22]: same as before -- it's not just due to identity resources

File: IdentityServer/v6/docs/content/fundamentals/claims.md [hunk 3]
```diff
@@ -7,63 +7,77 @@ weight: 45
...
-        return Task.CompletedTask;
+    private async Task<Claim> GetClaimsAsync(ProfileDataRequestContext context)
+    {
+        // Your implementation that retrieves claims goes here
+    }
+}
+```
+
+#### Always emit claims
+We generally recommend emitting claims based on the requested identity resources, as that respects the scopes requested by the client and gives the end user an opportunity to consent to this sharing of information. However, if you have claims that don't need to follow such rules, such as claims that are an integral part of the user's identity and that needed in most scenarios, they can be added by directly updating the *context.IssuedClaims* collection. For example:
```
Comment (brockallen) [inline IdentityServer/v6/docs/content/fundamentals/claims.md:52]: > requested identity resources

again, it's more the "requested claim types"

File: IdentityServer/v6/docs/content/fundamentals/claims.md (hunk 1 above)
Comment (josephdecock) [reply IdentityServer/v6/docs/content/fundamentals/claims.md]: Thanks for catching that! Fixed

**Review by brockallen — APPROVED** [review]

**Pull request merged**